From b0b8538e197b790bf2bffd77044d7a1176ea962e Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:13 +0800
Subject: [PATCH] arm64/mpam: monitor pmg as a property of partid
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b0b8538e197b790bf2bffd77044d7a1176ea962e


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

both match partid and pmg.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index 260490dad226..a33a01285d81 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -917,10 +917,11 @@ static void show_resctrl_tasks(struct rdtgroup *r, struct seq_file *s)
 	rcu_read_lock();
 	for_each_process_thread(p, t) {
 		if ((r->type == RDTCTRL_GROUP && t->closid == r->closid) ||
-		    (r->type == RDTMON_GROUP && t->rmid == r->mon.rmid))
-			seq_printf(s, "%d: partid = %d, pmg = %d, (group: partid %d, pmg %d)\n",
+		    (r->type == RDTMON_GROUP && t->closid == r->closid &&
+		     t->rmid == r->mon.rmid))
+			seq_printf(s, "%d: partid = %d, pmg = %d, (group: partid %d, pmg %d, mon %d)\n",
 				   t->pid, t->closid, t->rmid,
-				   r->closid, r->mon.rmid);
+				   r->closid, r->mon.rmid, r->mon.mon);
 	}
 	rcu_read_unlock();
 }
-- 
2.26.2

